src/realm/object-store/sync/app.cpp

```diff
@@ -257,26 +257,41 @@ void App::configure(const SyncClientConfig& sync_client_config)
 {
     auto sync_route = make_sync_route(m_app_route);
     m_sync_manager->configure(shared_from_this(), sync_route, sync_client_config);
-    m_logger = m_sync_manager->make_logger();
     if (auto metadata = m_sync_manager->app_metadata()) {
         update_hostname(metadata);
     }
 }
 
+inline bool App::init_logger()
+{
+    // If a log function is called before configure(), a null ptr will be
+    // returned by get_logger()
+    if (!m_logger_ptr) {
+        m_logger_ptr = m_sync_manager->get_logger();
```

Comment:
Does this return a nullptr if there's no logger configured? I.e. will every subsequent call try to get the logger if no logger factory is defined?

Reply:
It will only return a nullptr up until `App::configure()` is called - once that happens, either the logger factory will be used, if set, or a StdErrLogger will be created.